### ISSUE TITLE
Apply Humble object Pattern on Items

### ISSUE THREAD

**Issue by stilnat** [RE-SS3D/SS3D#1139]
## Summary

Currently, all the logic in items is contained in an actor class, making it troublesome to test for. It also prevent us to test containers correctly since containers are using items.

## Goal 

How I suggest to do it : 
- rename Item in ItemActor.
- Create a public Item nested class in it.
- Move all non network related logic from itemActor to Item.
- Duplicate public properties and methods of Item in ItemActor, like that : 
``` /// <summary>
/// The remaining volume in milliliters that fit in this container
/// </summary>
public float RemainingVolume => substanceContainer.RemainingVolume; 
```

Nested class allow Item to acces and update eventual private fields of ItemActor. This would be extra nice if it comes with a set of tests for the item class.

**Comment by TaylorNAlbarnaz** [comment]
It's more complex than it first sounded, you will have to manually change almost every single script that references items since they need both data from the Item like Name, Size, Weight; and from the ItemActor like Transform and Prefabs

**Comment by stilnat** [comment]
closed by #1140
